packages/DraftEditor/src/utils/index.tsx

```diff
@@ -1,5 +1,35 @@
 import ReactDOM from "react-dom";
 import { DEFAULT_HEIGHT, LEFT, MARGIN_IMAGE, RIGHT } from "../constants";
+import eventBus from "../eventBus";
+declare let window: any;
+const addButtonToAlignmentToolContainer = (element: any) => {
+  if (!element) {
+    return;
+  }
+  const editor = ReactDOM.findDOMNode(element);
+  if (editor instanceof HTMLElement) {
+    const [alignmentToolContainer] = Array.from(
+      editor.querySelectorAll(".draftJsEmojiPlugin__alignmentTool__2mkQr")
+    );
+
+    window.showModal = () => {
+      const [focusedImage] = Array.from(
+        editor.querySelectorAll(".draftJsFocusPlugin__focused__3Mksn")
+      );
+
+      eventBus.$emit("editImageAttributes", focusedImage.src);
+    };
+
+    alignmentToolContainer.insertAdjacentHTML(
+      "beforeend",
+      `<div class="draftJsEmojiPlugin__buttonWrapper__1Dmqh" onclick="showModal()">
+        <i class="material-icons">
+          perm_data_setting
+        </i>
+      </div>`
+    );
```

Comment:
is there no better way of doing this besideds inserting it directly?

Reply:
In this case I'm overwriting the alignmentTool plugin to insert a fragment that I need to handle